core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -0,0 +1,315 @@
+            .build();
+  }
+
+  public static ListTablesResponse listTables(Catalog catalog, Namespace namespace) {
+    List<TableIdentifier> idents = catalog.listTables(namespace);
+    return ListTablesResponse.builder().addAll(idents).build();
+  }
+
+  public static LoadTableResponse stageTableCreate(Catalog catalog, Namespace namespace, CreateTableRequest request) {
+    request.validate();
+
+    TableIdentifier ident = TableIdentifier.of(namespace, request.name());
+    if (catalog.tableExists(ident)) {
+      throw new AlreadyExistsException("Table already exists: %s", ident);
+    }
+
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put("created-at", OffsetDateTime.now().toString());
+    properties.putAll(request.properties());
+
+    TableMetadata metadata = TableMetadata.newTableMetadata(
+        request.schema(),
+        request.spec() != null ? request.spec() : PartitionSpec.unpartitioned(),
+        request.writeOrder() != null ? request.writeOrder() : SortOrder.unsorted(),
+        request.location(),
+        properties);
+
+    return LoadTableResponse.builder()
+        .withTableMetadata(metadata)
+        .build();
+  }
+
+  public static LoadTableResponse createTable(Catalog catalog, Namespace namespace, CreateTableRequest request) {
+    request.validate();
+
+    TableIdentifier ident = TableIdentifier.of(namespace, request.name());
+    Table table = catalog.buildTable(ident, request.schema())
+        .withLocation(request.location())
+        .withPartitionSpec(request.spec())
+        .withSortOrder(request.writeOrder())
+        .withProperties(request.properties())
+        .create();
+
+    if (table instanceof BaseTable) {
+      return LoadTableResponse.builder()
+          .withTableMetadata(((BaseTable) table).operations().current())
+          .build();
+    }
+
+    throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+  }
+
+  public static DropTableResponse dropTable(Catalog catalog, TableIdentifier ident) {
+    boolean dropped = catalog.dropTable(ident);
+    return DropTableResponse.builder().dropped(dropped).build();
+  }
+
+  public static LoadTableResponse loadTable(Catalog catalog, TableIdentifier ident) {
+    Table table = catalog.loadTable(ident);
+
+    if (table instanceof BaseTable) {
+      return LoadTableResponse.builder()
+          .withTableMetadata(((BaseTable) table).operations().current())
+          .build();
+    }
+
+    throw new IllegalStateException("Cannot wrap catalog that does not produce BaseTable");
+  }
+
+  public static LoadTableResponse updateTable(Catalog catalog, TableIdentifier ident, UpdateTableRequest request) {
+    TableMetadata finalMetadata;
+    if (isCreate(request)) {
+      // this is a hacky way to get TableOperations for an uncommitted table
+      Transaction transaction = catalog.buildTable(ident, EMPTY_SCHEMA).createOrReplaceTransaction();
```

Comment:
Is the plan here that we end up making a replace table transaction here which replaces the newly created table with the updates? I'm not quite sure I follow this branch

Reply:
Replace transactions are already working.

What's happening here is that the existing `Catalog` implementations don't allow you to get `TableOperations`. For all other commits, we load the table and grab `TableOperations` from `BaseTable` (see line 236). But for a create transaction, there is no table to load. The way I found to create `TableOperations` for a table that doesn't exist yet is to start a create transaction and get `TableOperations` from there.

As the comment says, this is a hacky way to get `TableOperations`, but I don't have a better one. Other options are: (1) don't support create transactions when wrapping `Catalog`, or (2) expose `TableOperations` in `Catalog` and force implementations to add a factory method. I think this is the cleanest way.